debezium-core/src/main/java/io/debezium/relational/ddl/AbstractDdlParser.java

```diff
@@ -0,0 +1,365 @@
+     * @param statement the DDL statement; may not be null
+     */
+    protected void signalCreateIndex(String indexName, TableId id, String statement) {
+        signalEvent(new DdlParserListener.TableIndexCreatedEvent(indexName, id, statement));
+    }
+
+    /**
+     * Signal a drop index event to all listeners.
+     *
+     * @param indexName the name of the index; may not be null
+     * @param id        the table identifier; may not be null
+     * @param statement the DDL statement; may not be null
+     */
+    protected void signalDropIndex(String indexName, TableId id, String statement) {
+        signalEvent(new DdlParserListener.TableIndexDroppedEvent(indexName, id, statement));
+    }
+
+    protected String removeLineFeeds(String input) {
+        return input.replaceAll("[\\n|\\t]", "");
+    }
+
+    /**
+     * Utility method to accumulate a parsing exception.
+     *
+     * @param e    the parsing exception
+     * @param list the list of previous parsing exceptions; may be null
+     * @return the list of previous and current parsing exceptions; if {@code e} is null then always {@code list}, but otherwise non-null list
+     */
+    protected Collection<ParsingException> accumulateParsingFailure(ParsingException e, Collection<ParsingException> list) {
+        if (e == null) return list;
+        if (list == null) list = new ArrayList<ParsingException>();
+        list.add(e);
+        return list;
+    }
+
+    /**
+     * Utility method to accumulate a parsing exception.
+     *
+     * @param e    the multiple parsing exceptions
+     * @param list the list of previous parsing exceptions; may be null
+     * @return the list of previous and current parsing exceptions; if {@code e} is null then always {@code list}, but otherwise non-null list
+     */
+    protected Collection<ParsingException> accumulateParsingFailure(MultipleParsingExceptions e, Collection<ParsingException> list) {
+        if (e == null) return list;
+        if (list == null) list = new ArrayList<ParsingException>();
+        list.addAll(e.getErrors());
+        return list;
+    }
+
+    protected Column createColumnFromConstant(String columnName, String constantValue) {
+        ColumnEditor column = Column.editor().name(columnName);
+        try {
+            if (constantValue.startsWith("'") || constantValue.startsWith("\"")) {
+                column.type("CHAR");
+                column.jdbcType(Types.CHAR);
+                column.length(constantValue.length() - 2);
+            } else if (constantValue.equalsIgnoreCase("TRUE") || constantValue.equalsIgnoreCase("FALSE")) {
+                column.type("BOOLEAN");
+                column.jdbcType(Types.BOOLEAN);
+            } else {
+                setTypeInfoForConstant(constantValue, column);
+            }
+        } catch (Throwable t) {
+            logger.debug("Unable to create an artificial column for the constant: " + constantValue);
```

Comment:
What is an artificial column?

Reply:
For now, I'm not sure with the answer. This is the code from existing implementation and i have no chance to use it at the moment